src/extensions/extension-discovery.ts

```diff
@@ -347,29 +337,40 @@ export class ExtensionDiscovery {
         isEnabled
       };
     } catch (error) {
-      logger.error(`${logModule}: can't install extension at ${manifestPath}: ${error}`, { manifestJson });
+      logger.error(`${logModule}: can't load extension manifest at ${manifestPath}: ${error}`, { manifestJson });
 
       return null;
     }
   }
 
-  async loadExtensions(): Promise<Map<LensExtensionId, InstalledExtension>> {
+  async ensureExtensions(): Promise<Map<LensExtensionId, InstalledExtension>> {
     const bundledExtensions = await this.loadBundledExtensions();
 
-    await this.installPackages(); // install in-tree as a separate step
-    const localExtensions = await this.loadFromFolder(this.localFolderPath);
+    await this.installPackages(this.packageJsonPath, bundledExtensions);
 
-    await this.installPackages();
-    const extensions = bundledExtensions.concat(localExtensions);
+    const userExtensions = await this.loadFromFolder(this.localFolderPath);
+    const extensions = bundledExtensions.concat(userExtensions);
 
-    return new Map(extensions.map(extension => [extension.id, extension]));
+    return this.extensions = new Map(extensions.map(extension => [extension.id, extension]));
   }
 
   /**
    * Write package.json to file system and install dependencies.
    */
-  installPackages() {
-    return extensionInstaller.installPackages(this.packageJsonPath, this.packagesJson);
+  async installPackages(packageJsonPath: string, extensions: InstalledExtension[]) {
```

Comment:
This could cause problems, if we first install bundled extensions then user extensions, the second install will remove the bundled extensions (as we have only 1 shared package.json and node_modules at the moment). The caller should ensure that the installed extensions are additive.

Reply:
We don't use this for installing user extensions.

Reply:
Maybe a change of function name then?